lib/search_solr_tools/logging/sst_logger.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'logging'
+require 'search_solr_tools'
+
+module SSTLogger
+  LOG_LEVELS = %w[debug info warn error fatal none].freeze
+
+  def logger
+    SSTLogger.logger
+  end
+
+  def log_environment(new_env)
+    SSTLogger.log_environment(new_env)
+  end
+
+  class << self
+    def logger
+      @logger ||= new_logger
```

Comment:
This is actually creating the logger in the SolrHarvestCLI class, right? I suggest skipping the trickiness and explicitly creating the logger as I note in a comment below. The approach here doesn't save any code because then you have to add the `log_environment` step. Make the logger setup easier to understand.

Reply:
The logger is used in many classes; having to pass that logger around from class to class would create a lot more complication than having it be used as a global.

Reply:
I must be missing something, then. It looks like each class that includes the logger module would add this logger method to itself, with each creating a new `@logger` instance. I'll look again.

Reply:
Each class does include the module.  The module basically uses it's "logger" method as a pass-through to the class method; the SSTLogger module/class are never instantiated, all the methods are class-level methods.

The module's logger method just passes the call to the class method, which creates a new logger if the @logger variable hasn't been created yet, or just passes back the existing one if it has.  They all share the same logger, though.

Reply:
Each class does include the Module, but the module itself is never instantiated; all the methods are class/module-level methods.

The "outer" `logger` method just allows us to call `logger.info` directly, instead of having to do `SSTLogging.logger`; that method basically just calls the "class" method called `logger`, which itself will create a new logger and assign it to `@logger`, unless the `@logger` class variable was already set (in which case, it just returns it).

So basically, the first time `logger.info` is called (by ANY class) it will instantiate the logger.  All subsequent logger calls will return the same already-instantiated logger, regardless of which class uses it.

Reply:
I'm not convinced of the wisdom of this approach, but I'll go with it.